flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/debezium/DebeziumJsonSerializationSchema.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.formats.json.debezium;
+
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.formats.json.JsonRowDataSerializationSchema;
+import org.apache.flink.formats.json.TimestampFormat;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.types.RowKind;
+
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static org.apache.flink.table.types.utils.TypeConversions.fromLogicalToDataType;
+
+/**
+ * Serialization schema from Flink Table/SQL internal data structure {@link RowData} to Debezium JSON.
+ *
+ * @see <a href="https://debezium.io/">Debezium</a>
+ */
+public class DebeziumJsonSerializationSchema implements SerializationSchema<RowData> {
+	private static final long serialVersionUID = 1L;
+
+	private static final StringData OP_INSERT = StringData.fromString("INSERT"); // insert
+	private static final StringData OP_DELETE = StringData.fromString("DELETE"); // delete
+
+	/** The serializer to serialize Debezium JSON data. **/
+	private final JsonRowDataSerializationSchema jsonSerializer;
+
+	private transient GenericRowData genericRowData;
+
+	public DebeziumJsonSerializationSchema(RowType rowType, TimestampFormat timestampFormat) {
+		jsonSerializer = new JsonRowDataSerializationSchema(
+			createJsonRowType(fromLogicalToDataType(rowType)),
+			timestampFormat);
+	}
+
+	@Override
+	public void open(InitializationContext context) {
+		genericRowData = new GenericRowData(2);
+	}
+
+	@Override
+	public byte[] serialize(RowData rowData) {
+		try {
+			genericRowData.setField(0, rowData);
```

Comment:
This is wrong. Debezium serialize insert data in "after", but delete data in "before".